std/datetime.d

```diff
@@ -1075,6 +1075,25 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        if (_stdTime < 0)
+            return (cast(size_t) (_stdTime * -1)) + long.max;
```

Comment:
for 64 bit machines you can just return `_stdTime` without any magic and collisions.
For 32 bit machines your code does not work anyway